### DISCUSSION TITLE
[wtpy][cta引擎] 使用执行器时，当更改发单代码后，仍然会创建上一个版本代码的执行器并执行

### DISCUSSION THREAD

**Discussion by ZzzzHeJ** [wondertrader/wondertrader#61] [Q&A] [answered]
使用执行器时，当更改发单代码后，仍然会创建上一个版本代码的执行器并执行，并且导致当前线程无法发单。

**Comment by delusionpig** [comment]
由于上一个代码的单没有成功发出去，默认延后执行，vscode里面debug模式下直接停止并没有实际结束该进程，下次执行时还是判断由于上一个单没有发出然后继续阻塞，我试了一下使用控制台结束任务再重新启动任务后就会顺延执行阻塞的任务

**Reply by wondertrader** [reply]
> 由于上一个代码的单没有成功发出去，默认延后执行，vscode里面debug模式下直接停止并没有实际结束该进程，下次执行时还是判断由于上一个单没有发出然后继续阻塞，我试了一下使用控制台结束任务再重新启动任务后就会顺延执行阻塞的任务

那是不是说已经解决了？

**Comment by ZzzzHeJ** [comment, marked as answer]
**解决** 是因为在策略运行中，人工干预平掉了持仓，导致generated中理论持仓与实际持仓不一致。wt会将实际持仓调整为理论持仓一致，导致出现幽灵订单
